Kickstarter-iOS/Features/SortFilter/FilterCategory/FilterCategoryView.swift

```diff
@@ -74,23 +77,59 @@ struct FilterCategoryView<T: FilterCategory>: View {
             .foregroundStyle(Colors.Text.primary.swiftUIColor())
           Spacer()
           self.radioButton(isSelected: self.viewModel.isCategorySelected(category))
+            .id("\(category.id)-radio-button")
         }
+        .background(Colors.Background.surfacePrimary.swiftUIColor())
         .padding(.vertical, Constants.rowPaddingVertical)
         .padding(.horizontal, Constants.rowPaddingHorizontal)
 
+        self.subcategories(
+          for: category
+        )
+
         self.separator
+          .id("\(category.id)-separator")
       }
       .background(Colors.Background.surfacePrimary.swiftUIColor())
       .listRowInsets(EdgeInsets()) // Remove List internal insets
       .listRowSeparator(.hidden) // Hide default separators
       .contentShape(Rectangle())
       .onTapGesture { [weak viewModel] () in
-        viewModel?.selectCategory(category)
+        if let viewModel, !viewModel.isCategorySelected(category) {
+          viewModel.selectCategory(category)
+        }
       }
+      .id(category.id)
     }
     .listStyle(.plain)
   }
 
+  @ViewBuilder
+  private func subcategories(for category: T) -> some View {
+    if self.viewModel.isCategorySelected(category), let subcategories = category.availableSubcategories {
+      FlowLayout(horizontalSpacing: 8, verticalSpacing: 8, alignment: .leading) {
+        TitlePillButton(
+          title: Strings.Project_status_all(),
+          isHighlighted: self.viewModel.isSubcategorySelected(nil),
+          count: nil
+        ) {
+          self.viewModel.selectCategory(category, subcategory: nil)
+        }
+        ForEach(subcategories) { subcategory in
+          TitlePillButton(
+            title: subcategory.name,
+            isHighlighted: self.viewModel.isSubcategorySelected(subcategory),
+            count: nil
+          ) {
+            self.viewModel.selectCategory(category, subcategory: subcategory)
+          }
+          .id(subcategory.id)
+        }
+      }
+      .padding(EdgeInsets(top: 4, leading: 24, bottom: 16, trailing: 24))
```

Comment:
nit: maybe not super useful since this is all contained in this ViewBuilder, but a constant to keep track of these values (as well as the FlowLayout spacing values above) might be helpful in the future.